programs/x509/cert_write.c

```diff
@@ -207,6 +216,18 @@ struct options {
     int format;                 /* format                               */
 } opt;
 
+static void ip_string_to_bytes(const char *str, uint8_t *bytes, int maxBytes)
+{
+    for (int i = 0; i < maxBytes; i++) {
+        bytes[i] = (uint8_t) strtoul(str, NULL, 16);
+        str = strchr(str, '.');
+        if (str == NULL || *str == '\0') {
+            break;
+        }
+        str++;
+    }
+}
+
```

Comment:
IPv4 addresses are usually decimal, but I recommend against trying to parse them manually.  Instead, use `getaddrinfo()`  from the C standard library, with `AI_NUMERICHOST` and `AI_NUMERICSERV` to prevent DNS lookups.

Reply:
Please see the comment above.